Character.cpp

```diff
@@ -98,20 +98,32 @@ void Character::attackInternal(Character& other)
             b) your stats are boosted 10%
             c) the initial value of your stats is updated to reflect this boosted stat for the next time you defeat another character.
       */
-        assert(false);
-        std::cout << getName() << " defeated " << other.getName() << " and leveled up!" << std::endl;        
+        std::cout << getName() << " defeated " << other.getName() << " and leveled up!" << std::endl; 
+        defeatOther(*initialHitPoints, hitPoints, 0.5f);
+        defeatOther(*initialArmorLevel, armor, 0.25f);
+        defeatOther(*initialAttackDamage, attackDamage, 0.1f);
```

Comment:
The 3rd parameter is not necessary here and is unused in the function implementation.
remove it.
- [ ] revise